msteams-platform/tabs/how-to/authentication/auth-flow-tab.md

```diff
@@ -6,42 +6,48 @@ keywords: teams authentication flow tabs
+Similar to other application auth flows in Teams, the start page must be on a domain that is in its `validDomains` list, and on the same domain as the post sign in redirect page.
+
+    > [!NOTE]
+    > The OAuth 2.0 implicit grant flow calls for a `state` parameter in the authentication request, which contains unique session data to prevent a [cross-site request forgery attack](https://en.wikipedia.org/wiki/Cross-site_request_forgery). The examples use a randomly-generated GUID for the `state` data.
+
 5. On the provider's site, the user signs in and grants access to the tab.
 6. The provider takes the user to the tab's OAuth 2.0 redirect page with an access token.
 7. The tab checks that the returned `state` value matches what was saved earlier, and calls `microsoftTeams.authentication.notifySuccess()`, which in turn calls the `successCallback` function registered in step 3.
 8. Teams closes the pop-up window.
-9. The tab either displays configuration UI or refreshes or reloads the tabs content, depending on where the user started from.
+9. The tab either displays configuration UI, refreshes, or reloads the tabs content, depending on where the user started from.
 
 ## Treat tab context as hints
 
-Although the tab context provides helpful information regarding the user, don't use this information to authenticate the user whether you get it as URL parameters to your tab content URL or by calling the `microsoftTeams.getContext()` function in the Microsoft Teams client SDK. A malicious actor could invoke your tab content URL with its own parameters, and a web page impersonating Microsoft Teams could load your tab content URL in an iframe and return its own data to the `getContext()` function. You should treat the identity-related information in the tab context simply as hints and validate them before use. Refer to the notes in [Navigate to the authorization page from your popup page](~/tabs/how-to/authentication/auth-tab-aad.md#navigate-to-the-authorization-page-from-your-popup-page).
+Although the tab context provides helpful information regarding the user, do not use this information to authenticate the user. Do not authenticate the user even if you get the information as URL parameters to your tab content URL or by calling the `microsoftTeams.getContext()` function in the Microsoft Teams client SDK. A malicious actor can invoke your tab content URL with its own parameters. The actor can also invoke a web page impersonating Microsoft Teams to load your tab content URL in an iframe and return its own data to the `getContext()` function. You must treat the identity-related information in the tab context simply as hints and validate them before use. Refer to the notes in [navigate to the authorization page from your pop-up page](~/tabs/how-to/authentication/auth-tab-aad.md#navigate-to-the-authorization-page-from-your-popup-page).
```

Comment:
This alters the meaning here. Idea here is to get user authenticated even if you could get some information easily via getContext() to avoid attacks. 

So, please change "Do not authenticate the user even if"  =>  "Do authenticate the user even if..."